internal/pkg/earpointer/taint.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package earpointer
```

Comment:
IMO something is off here. earpointer is a different set of concerns than taint. Taint on top of EAR is a more natural split. So I kinda suspect this is not the long term home for this code. I'll let @mlevesquedion and @PurelyApplied make a judgement call if they are okay with this.

My real concern is are breaking the abstraction between these by combining them into the same package.

Reply:
I can introduce a separate package "eartaint" for this. I have no preference though.